bin/faucet/src/static/index.js

```diff
@@ -34,19 +45,30 @@ document.addEventListener('DOMContentLoaded', function () {
         }
 
         button.textContent = 'Loading...';
+        infoContainer.style.display = 'none';
+        importCommand.style.display = 'none';
+
         try {
             const response = await fetch('http://localhost:8080/get_tokens', {
                 method: 'POST',
                 headers: { 'Content-Type': 'application/json' },
-                body: JSON.stringify({ account_id: accountId })
+                body: JSON.stringify({ account_id: accountId, is_private_note: isPrivateNote})
             });
 
             if (!response.ok) {
                 throw new Error(`HTTP error! Status: ${response.status}`);
             }
 
             const blob = await response.blob();
-            downloadBlob(blob, 'note.mno');
+            if(isPrivateNote) {
+                importCommand.style.display = 'block';
+                downloadBlob(blob, 'note.mno');
+            }
+
+            const noteId = response.headers.get('Content-Disposition').split('filename=')[1].replace(/"/g, '');
```

Comment:
This looks a bit complicated but also it might be fine for now. I guess an alternative would be for the name could be sent directly through a custom header to make it a bit simpler.
Alternatively the server could return a JSON response that contains the type of response, and (depending on the user selection) the encoded note details.

Reply:
Done! Didn't know about custom headers.